### PR TITLE
Fix broken doc links and enforce via mdbook-linkcheck plugin

### PR DESCRIPTION
**Pull request by rukai**
The setup script is adjusted to run for both netlify and local usage (previously just local usage)

I opted to delete most of the dead-links but im happy to comment them out if we want to keep them around until we bring back the pages they link to.

### CODE REVIEW

File: docs/mdbook.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e; set -u
+
+# Setup the the mdbook binaries if they havent already been setup
```
Comment (conorbros) [inline docs/mdbook.sh:5]: Can we install with cargo?

Comment (rukai) [reply docs/mdbook.sh:5]: It needs to run on netlify so unfortunately not.

Comment (conorbros) [reply docs/mdbook.sh:5]: Can we do the checks on our CI instead?

Comment (rukai) [reply docs/mdbook.sh:5]: Doing that would mean we have two locations that we need to manage running mdbook.
Surely doing it all in one place would be simpler?

If we build both mdbook and mdbook-linkcheck on CI that will make CI run longer and use more of our cache.

Comment (rukai) [reply docs/mdbook.sh:5]: There are some basic syntax checks that netlify is already handling for us.

**Comment by rukai** [comment]
Currently blocked on https://github.com/Michael-F-Bryan/mdbook-linkcheck/pull/65
Will split out the link fixes if the issue takes too long to resolve.

**Comment by rukai** [comment]
Oh wow our netlify image is on xenial (16.04) 
image

@benbromhead I dont seem to be able to set this from the config file.
Can you follow the instructions to upgrade the image here: https://answers.netlify.com/t/new-ubuntu-20-04-focal-fossa-build-image/37095

Should be completely safe as everything is statically linked (except for libc)

**Comment by benbromhead** [comment]
Done!

**Comment by rukai** [comment]
It works!

**Review by conorbros — APPROVED** [review]
LGTM!

**Review by benbromhead — APPROVED** [review]

**Pull request merged**